Sources/StreamChat/APIClient/RequestEncoder.swift

```diff
@@ -45,16 +45,23 @@ extension RequestEncoder {
             subsystems: .httpRequests
         )
 
-        var result: Result<URLRequest, Error>?
-        encodeRequest(for: endpoint) { result = $0 }
-
-        log.assert(
-            result != nil,
-            "`encodeRequest` with `requiresConnectionId == false` should return immediately.",
-            subsystems: .httpRequests
+        var result: Result<URLRequest, Error> = .failure(
+            ClientError("Unexpected error. The result was not changed after encoding the request.")
         )
 
-        return try result!.get()
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        encodeRequest(for: endpoint) {
+            result = $0
+            dispatchGroup.leave()
+        }
+
+        let waitResult = dispatchGroup.wait(timeout: .now() + 5)
```

Comment:
Why 5 seconds? Isn't it too much?

Reply:
This is not going to happen ever. That is why I put a random value. But just as safe percussion, I put a timeout. But makes sense to extract this to a variable or just use the timeout from the ClientConfig. I'll improve this 👍